edgelet/iotedged/src/lib.rs

```diff
@@ -1442,25 +1493,70 @@ where
         runt_rx,
     )?;
 
+    // This mpsc sender/receiver is used for getting notifications from the mgmt service
+    // indicating that the daemon should shut down and attempt to reprovision the device.
+    let mgmt_stop_and_reprovision_signaled = mgmt_stop_and_reprovision_rx
+        .then(|res| match res {
+            Ok(_) => Err(None),
+            Err(_) => Err(Some(Error::from(ErrorKind::ManagementService))),
+        })
+        .for_each(move |_x: Option<Error>| Ok(()))
+        .then(|res| match res {
+            Ok(_) | Err(None) => Ok(None),
+            Err(Some(e)) => Err(Some(e)),
+        });
+
+    let mgmt_stop_and_reprovision_signaled =
+        if let Some(&true) = settings.provisioning().dynamic_reprovisioning() {
+            futures::future::Either::B(mgmt_stop_and_reprovision_signaled)
+        } else {
+            futures::future::Either::A(future::empty().map(|_: ()| None))
```

Comment:
`.map(|_: ()| None)` -> `.map(|()| None)` , ie use `()` as a pattern.

Reply:
Actually, do you really need the `.map` at all? I would assume the type parameter of `empty()` would infer to `Option` already. Even if it doesn't, you can turbofish it.

Reply:
You're right, removed the map.